src/gfxinit.cpp

```diff
@@ -309,7 +331,7 @@ static bool SwitchNewGRFBlitter()
 		if (BlitterFactory::GetBlitterFactory(repl_blitter) == nullptr) continue;
 
 		/* Inform the video driver we want to switch blitter as soon as possible. */
-		VideoDriver::GetInstance()->ChangeBlitter(repl_blitter);
+		VideoDriver::GetInstance()->QueueOnMainThread(std::bind(&RealChangeBlitter, repl_blitter));
```

Comment:
Shouldn't this also be a Lambda similar as screenshot?

Reply:
That is purely a matter of style opinion. Formulating this with a lambda is just as possible, but there's not any functional difference.